### PR TITLE
macvim 8.2-172

### PR DESCRIPTION
**Pull request by moonfruit**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
Looks like this tries to build universal binaries by default, and also installs pre-built binaries. We should try to disable building universal binaries, and try to build the pre-built ones from source.

**Comment by moonfruit** [comment]
I test this on my x86_64 macOS, it does not create universal binaries. @carlocab

**Comment by carlocab** [comment]
Did you test this in `brew`, or did you compile outside of Homebrew? You should try the latter.

This also installs a pre-built `Sparkle.framework`, which is a universal binary:
```
❯ file src/MacVim/Sparkle.framework/Sparkle
src/MacVim/Sparkle.framework/Sparkle: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [arm64:Mach-O 64-bit dynamically linked shared library arm64]
src/MacVim/Sparkle.framework/Sparkle (for architecture x86_64): Mach-O 64-bit dynamically linked shared library x86_64
src/MacVim/Sparkle.framework/Sparkle (for architecture arm64):  Mach-O 64-bit dynamically linked shared library arm64
```

**Comment by moonfruit** [comment]
I test it in `brew` just now.

**Comment by moonfruit** [comment]
And I try it again:

```console
$ file /usr/local/Cellar/macvim/8.2-172/MacVim.app/Contents/MacOS/Vim
/usr/local/Cellar/macvim/8.2-172/MacVim.app/Contents/MacOS/Vim: Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE>
$ file /usr/local/Cellar/macvim/8.2-172/MacVim.app/Contents/MacOS/MacVim
/usr/local/Cellar/macvim/8.2-172/MacVim.app/Contents/MacOS/MacVim: Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE>
```

**Comment by carlocab** [comment]
It seems to install it anyway, along with some other things that look pre-built:
```
==> brew audit macvim --online --git --skip-style
==> FAILED
macvim:
  * Unexpected universal binaries were found.
    The offending files are:
      /usr/local/Cellar/macvim/8.2-172/MacVim.app/Contents/Frameworks/Sparkle.framework/Versions/A/Resources/Autoupdate.app/Contents/MacOS/Autoupdate
      /usr/local/Cellar/macvim/8.2-172/MacVim.app/Contents/Frameworks/Sparkle.framework/Versions/A/Resources/Autoupdate.app/Contents/MacOS/fileop
      /usr/local/Cellar/macvim/8.2-172/MacVim.app/Contents/Frameworks/Sparkle.framework/Versions/A/Sparkle
Error: 1 problem in 1 formula detected
```
The build also fails on Big Sur. Probably due to Xcode 13.

**Comment by moonfruit** [comment]
It does contain `Sparkle.framework`. I didn't note that. But when build with `--disable-sparkle`, it is safe to delete `Sparkle.framework`.
And on my x86_64 MacBook, there are both `Xcode 13` and `CommandLineTools for Xcode 13`. I don't know which one is used when brew builds the formula. But it did build successfully.
I don't know why it failed on arm64 because I don't have such an environment.

File: Formula/macvim.rb [hunk 1]
```diff
@@ -55,6 +54,7 @@ def install
                           "--with-macarchs=#{Hardware::CPU.arch}"
     system "make"
 
+    rm_rf Dir["src/MacVim/build/Release/MacVim.app/Contents/Frameworks/Sparkle.framework"]
```
Comment (carlocab) [inline Formula/macvim.rb:57]:
```suggestion
    (buildpath/"src/MacVim/build/Release/MacVim.app/Contents/Frameworks/Sparkle.framework").rmtree
```
Also, we should move this to the start of the `install` method.

**Comment by carlocab** [comment]
> I don't know why it failed on arm64 because I don't have such an environment.

It failed on Intel Big Sur in CI also.

**Comment by moonfruit** [comment]
But on my Intel Big Sur, it is success.

**Comment by carlocab** [comment]
The ARM build succeeded too before the runners were upgraded to Xcode 13.

**Pull request closed without merging**